lib/Console/Application.php

```diff
@@ -22,22 +22,41 @@
 
 use \Symfony\Component\Console\Input\InputInterface;
 use \Symfony\Component\Console\Output\OutputInterface;
+use \Symfony\Component\Console\ConsoleEvents;
+use \Symfony\Component\EventDispatcher\EventDispatcher;
+use \Symfony\Component\Console\Event\ConsoleCommandEvent;
 
 class Application extends \Symfony\Component\Console\Application
 {
     public function __construct()
     {
         parent::__construct('Mesamatrix CLI', \Mesamatrix::$config->getValue('info', 'version'));
+
+        $dispatcher = new EventDispatcher();
+
+        // Install CLI logger
+        $dispatcher->addListener(ConsoleEvents::COMMAND,
+            function (ConsoleCommandEvent $e)
+            {
+                $logger = new \Monolog\Logger('cli.'.$e->getCommand()->getName());
+                foreach (\Mesamatrix::$logger->getHandlers() as $handler) {
+                    $logger->pushHandler($handler);
+                }
+                $logger->pushHandler(
+                    new \Symfony\Bridge\Monolog\Handler\ConsoleHandler($e->getOutput())
+                );
+                \Mesamatrix::$logger = $logger;
```

Comment:
OK, I know what is bothering me here (on top of the fact that I don't like letting the garbage collector cleaning our mess): you're basically copying the Mesamatrix logger into the newly created one, only to reinject it in Mesamatrix.

As I tried to make a cleaner version, I figured out that my main question is: why using the dispatcher?